operators/src/processing/point_in_polygon.rs

```diff
@@ -0,0 +1,775 @@
+
+impl InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>
+    for InitializedPointInPolygonFilter
+{
+    fn query_processor(&self) -> Result<TypedVectorQueryProcessor> {
+        let point_processor = if let TypedVectorQueryProcessor::MultiPoint(point_processor) =
+            self.vector_sources[0].query_processor()?
+        {
+            point_processor
+        } else {
+            unreachable!("checked in `PointInPolygonFilter` constructor");
+        };
+
+        let polygon_processor = if let TypedVectorQueryProcessor::MultiPolygon(polygon_processor) =
+            self.vector_sources[1].query_processor()?
+        {
+            polygon_processor
+        } else {
+            unreachable!("checked in `PointInPolygonFilter` constructor");
+        };
+
+        Ok(TypedVectorQueryProcessor::MultiPoint(
+            PointInPolygonFilterProcessor::new(point_processor, polygon_processor).boxed(),
+        ))
+    }
+}
+
+pub struct PointInPolygonFilterProcessor {
+    points: Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
+    polygons: Box<dyn VectorQueryProcessor<VectorType = MultiPolygonCollection>>,
+}
+
+impl PointInPolygonFilterProcessor {
+    pub fn new(
+        points: Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
+        polygons: Box<dyn VectorQueryProcessor<VectorType = MultiPolygonCollection>>,
+    ) -> Self {
+        Self { points, polygons }
+    }
+
+    fn filter_points(
+        points: &MultiPointCollection,
+        polygons: &MultiPolygonCollection,
+        initial_filter: &BooleanArray,
+    ) -> Result<BooleanArray> {
+        let mut filter = Vec::with_capacity(points.len());
+
+        let tester = PointInPolygonTester::new(polygons);
+
+        let coordinates = points.coordinates();
+
+        for ((&coordinates_start_index, &coordinates_end_index), time_interval) in points
+            .multipoint_offsets()
+            .iter()
+            .tuple_windows()
+            .zip(points.time_intervals())
+        {
+            let (coordinates_start_index, coordinates_end_index) = (
+                coordinates_start_index as usize,
+                coordinates_end_index as usize,
+            );
+
+            let is_multi_point_in_polygon_collection = coordinates
+                [coordinates_start_index..coordinates_end_index]
+                .iter()
+                .any(|coordinate| tester.is_coordinate_in_collection(coordinate, time_interval));
+
+            filter.push(is_multi_point_in_polygon_collection);
+        }
+
+        arrow::compute::or(initial_filter, &filter.into()).map_err(Into::into)
+    }
+}
+
+impl VectorQueryProcessor for PointInPolygonFilterProcessor {
+    type VectorType = MultiPointCollection;
+
+    fn vector_query(
+        &self,
+        query: QueryRectangle,
+        ctx: QueryContext,
+    ) -> BoxStream<'_, Result<Self::VectorType>> {
+        // TODO: multi-threading
```

Comment:
must be cautious when using simd AND multithreading. not all cores have simd units so might get slower the more threads we use.

Reply:
Yes, but you can compute the boolean vector in parallel and then use simd for `or`.